cloud_info/providers/openstack.py

```diff
@@ -211,7 +227,135 @@ def _get_dn(self, x509name):
+                                                     header_server).group(1)
+                    e_version = re.search(r'OCCI/([0-9.]+)',
+                                          header_server).group(1)
+            except requests.exceptions.RequestException:
+                pass
+            except IndexError:
+                pass
+        elif endpoint_type == 'compute':
+            try:
+                # TODO(gwarf) Retrieve using API programatically
+                e_version = urlparse(endpoint_url).path.split('/')[1]
+            except Exception:
+                pass
+
+        ret.update({
+            'compute_middleware_version': e_middleware_version,
+            'compute_api_version': e_version,
+        })
+
+        return ret
+
+    def _get_endpoint_ca_information(self, endpoint_url, insecure, cacert):
+        '''Return the cer issuer and trusted CAs list of an HTTPS endpoint.'''
+        ca_info = {'issuer': 'UNKNOWN', 'trusted_cas': ['UNKNOWN']}
+
+        if insecure:
+            verify = SSL.VERIFY_NONE
+        else:
+            verify = SSL.VERIFY_PEER
+
+        try:
+            scheme = urlparse(endpoint_url).scheme
+            host = urlparse(endpoint_url).hostname
+            port = urlparse(endpoint_url).port
+
+            if scheme == 'https':
+                ctx = SSL.Context(SSL.SSLv23_METHOD)
+                ctx.set_options(SSL.OP_NO_SSLv2)
+                ctx.set_options(SSL.OP_NO_SSLv3)
+                ctx.set_verify(verify, lambda conn, cert, errno, depth, ok: ok)
+                if not insecure:
+                    ctx.load_verify_locations(cacert)
+
+                client = socket.socket(socket.AF_INET, socket.SOCK_STREAM)
+                client.connect((host, port))
+
+                client_ssl = SSL.Connection(ctx, client)
+                client_ssl.set_connect_state()
+                client_ssl.set_tlsext_host_name(host)
+                client_ssl.do_handshake()
+
+                cert = client_ssl.get_peer_certificate()
+                issuer = self._get_dn(cert.get_issuer())
+
+                client_ca_list = client_ssl.get_client_ca_list()
+                trusted_cas = [self._get_dn(ca) for ca in client_ca_list]
+
+                client_ssl.shutdown()
+                client_ssl.close()
+
+                ca_info['issuer'] = issuer
+                ca_info['trusted_cas'] = trusted_cas
+        except SSL.Error:
+            pass
+
+        return ca_info
+
+    def _get_dn(self, x509name):
+        '''Return the DN of an X509Name object.'''
+        # XXX shortest/easiest way to get the DN of the X509Name
+        # str(X509Name): <X509Name object 'DN'>
+        # return str(x509name)[18:-2]
+        obj_name = str(x509name)
+        start = obj_name.find("'") + 1
+        end = obj_name.rfind("'")
+
+        return obj_name[start:end]
+
+    @_rescope
+    def get_compute_endpoints(self, os_project_name=None, **kwargs):
```

Comment:
@alvarolopez this change (and the similar ones on `get_templates` and `get_images` are not coherent with the base class:
https://github.com/EGI-Foundation/cloud-info-provider/blob/d07d879094e631c26e0f051eaa9721f8fa7139a2/cloud_info/providers/__init__.py#L17
 Is there any clean way to easily handle this?